### PR TITLE
feat(app switcher): add loading, failure, and empty states

### PR DESCRIPTION
**Pull request by corykon**
re #1044

@joeskeen This'll likely cause some conflicts with your work! Sorry! Feel free to grab me to work on merging if needed.

**Empty State:**
appSwitcher-empty

**Loading and Fail State:**
appSwitcher-fail

For testing, you can make tweaks in `...Fabric.Cashmere\projects\cashmere-examples\src\lib\navbar-app-switcher\custom-app-switcher.service.ts`

```
getApplications(): Observable<IDiscoveryRequest> {
    return of({
        value: []
    }).pipe(delay(5000)).pipe(tap(() => { throw Error("hi"); }));
}
```

### CODE REVIEW

**Comment by joeskeen** [comment]
@corykon Do we no longer have a View All My Applications URL?  I would think that should be available even on error or empty states if configured.

File: projects/cashmere/src/lib/app-switcher/app-switcher.module.ts [hunk 1]
```diff
@@ -4,22 +4,25 @@ import {HttpClientModule} from '@angular/common/http';
 
 import {AppSwitcherComponent} from './app-switcher.component';
 import {PopModule} from '../pop/popover.module';
+import {ProgressIndicatorsModule} from '../progress-indicators/progress-indicators.module';
 import {AppSwitcherService} from './app-switcher.service';
+import { WorkTrackerService } from '../picklist/services/work-tracker.service';
```
Comment (joeskeen) [inline projects/cashmere/src/lib/app-switcher/app-switcher.module.ts]: ...so app switcher is now dependent on picklist?  Seems strange to me...

Comment (corykon) [reply projects/cashmere/src/lib/app-switcher/app-switcher.module.ts]: Just a helper service that I didn't realize it was sitting within the picklist directory. I can move it to a more general location if you'd like.

Comment (joeskeen) [reply projects/cashmere/src/lib/app-switcher/app-switcher.module.ts]: Maybe this service should belong someplace else if it is used in multiple places

**Comment by corykon** [comment]
> @corykon Do we no longer have a View All My Applications URL? I would think that should be available even on error or empty states if configured.

@andrew-frueh would you agree that I ought to add the link to the empty and error states? While I'm at it, could I move the link up top? Sometimes that right aligned at the bottom feels a little off, I think because of the center aligned feeling of the app icons.

This would be the tweaked version:
image

File: projects/cashmere/src/lib/app-switcher/app-switcher.module.ts (hunk 1 above)
Comment (corykon) [reply projects/cashmere/src/lib/app-switcher/app-switcher.module.ts]: @joeskeen do you think `...\Fabric.Cashmere\projects\cashmere\src\lib\shared` would be an ok place to move `WorkTrackerService`?

**Comment by andrew-frueh** [comment]
I worry that the View All at the top might be a little less findable, but aesthetically I agree it's a big improvement.   So let's do it.  Just double check your vertical alignment on the view all button.  Right now it's aligned with the descender on the 'p' of Applications, so it's a little off.  Could either nudge it up 3px or so, or maybe `align-items: center` if that looks better

**Comment by andrew-frueh** [comment]
Also @corykon - you could probably tighten up the padding-bottom on that version with View All at the top.  There's more white-space than we need below the thumbnails in that example.

**Comment by corykon** [comment]
@andrew-frueh Agreed on all points. That was just an image from adobe where I covered up the bottom link and quickly threw a new one up top. When I do it for real I'll be sure it's tight.

**Comment by corykon** [comment]
@andrew-frueh here's the real deal. Feedback before I merge?
image

**Comment by andrew-frueh** [comment]
Looking great @corykon.  Hey can you add one css line while you're in there, below line 13 in app-switcher.component.scss - can you add `padding-left: 2px`?  For some reason "Applications" looks like it's falling off the left edge of the line below it.  That's not something new - it's been that way, but I just noticed it now.

**Comment by andrew-frueh** [comment]
Thanks @corykon - looks like you're good to merge your branch

**Pull request merged**